### PR TITLE
feat(878): implement network txn toggle and new style

### PR DESCRIPTION
**Pull request by DDDDDanica**
## Explanation
Fix: https://github.com/MetaMask/MetaMask-planning/issues/878

- Divide the `show all transaction` to individual toggles for each network to decide the preference respectfully.
- Apply such flow for both **onboarding** and **settings** page
- Show all toggles button on the right side of content
- Migrate `preferences.test` from mocha to jest

## Screenshots/Screencaps

https://github.com/MetaMask/metamask-extension/assets/12678455/7438049f-fb62-4dc0-92c2-4fb4c5c3de59

### Before
**Security & privacy page**
Screenshot 2023-08-02 at 13 48 24
**Advanced page**
Screenshot 2023-08-02 at 13 48 33

### After
**Security & privacy page**
Screenshot 2023-08-02 at 13 42 48

**Advanced page**
Screenshot 2023-08-02 at 13 43 08

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.test.js [hunk 1]
```diff
@@ -25,6 +25,7 @@ describe('Snap Permission List', () => {
   it('renders permissions list for snaps', () => {
     renderWithProvider(
       <SnapPermissionsList
+        snapId="snapid"
```
Comment (brad-decker) [inline ui/components/app/snaps/snap-permissions-list/snap-permissions-list.test.js:28]: Is this related to your changes?

File: ui/components/ui/toggle-button/index.scss [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .toggle-button {
   display: flex;
   cursor: pointer;
-
+  text-align: center;
```
Comment (brad-decker) [inline ui/components/ui/toggle-button/index.scss:4]: This file is used pretty globally, are we sure that this doesn't impact other toggles display?

**Review by brad-decker — DISMISSED** [review]
I don't know for sure but I think there may have been some refactors not related to these changes mixed in with the effort to update the toggles. It seems like changes were made to parts of the settings screens unrelated to the incoming transaction preferences which is very much needed and wanted but is making evaluating the effort of multiplexing the incoming tx toggles difficult. How complex would it be to pull the changes to components (Typography to Text, divs to Box, etc) into a PR that proceeds this one?

File: ui/components/app/incoming-trasaction-toggle/hooks.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  allNetworks,
+  incomingTransactionsPreferences,
+  setIncomingTransactionsPreferences,
+}) => {
+  const [networkPreferences, setNetworkPreferences] = useState(
+    generateIncomingNetworkPreferences(
+      incomingTransactionsPreferences,
+      allNetworks,
+    ),
+  );
```
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/hooks.js:14]: >Are we showing a toggle for every network they have added
Yes, indeed. 

>The incoming transactions controller only supports the networks in the ETHERSCAN_SUPPORTED_NETWORKS constant
````
    const { incomingTransactionsPreferences } =
      this.preferencesController.store.getState();
   const hasIncomingTransactionsFeatureEnabled = Boolean(
      incomingTransactionsPreferences[chainId],
    );
```
i believe this would prevent the issue, wdyt?

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.test.js (hunk 1 above)
Comment (DDDDDanica) [reply ui/components/app/snaps/snap-permissions-list/snap-permissions-list.test.js:28]: it was a fix. but will remove for now 🙏🏻

File: ui/components/app/incoming-trasaction-toggle/hooks.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from 'react';
+import { TEST_CHAINS } from '../../../../shared/constants/network';
+
+export const useIncomingTransactionToggle = ({
```
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/hooks.js:4]: Originally i wanted to separate it to make the file smaller and more readable. easier to manage as well. If i change the name to `useIncomingTransactionToggleHook`, could it be a compromised solution?

**Comment by DDDDDanica** [comment]
Hey @brad-decker thanks for reviewing! it's a good idea to separate the component changes, let me try it ~

File: ui/components/ui/toggle-button/index.scss (hunk 2 above)
Comment (DDDDDanica) [reply ui/components/ui/toggle-button/index.scss:4]: this is only used for <ToggleButton> component so this is what we wanted ~

File: ui/pages/onboarding-flow/privacy-settings/index.scss [hunk 5]
```diff
@@ -23,9 +23,7 @@
 
   &__settings {
     display: flex;
-    justify-content: center;
     flex-direction: column;
-    text-align: center;
```
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply ui/pages/onboarding-flow/privacy-settings/index.scss:28]: This is to remove the center for header 
Before:
Screenshot 2023-08-02 at 18 05 02

After:
Screenshot 2023-08-02 at 18 05 16

File: ui/components/app/incoming-trasaction-toggle/hooks.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/hooks.js]: see [1ba6cb1f7d489bea929d9b6969e964406b94543e](https://github.com/MetaMask/metamask-extension/pull/20363/commits/1ba6cb1f7d489bea929d9b6969e964406b94543e)

File: ui/components/app/snaps/snap-permissions-list/snap-permissions-list.test.js (hunk 1 above)
Comment (DDDDDanica) [reply ui/components/app/snaps/snap-permissions-list/snap-permissions-list.test.js]: see [1ba6cb1f7d489bea929d9b6969e964406b94543e](https://github.com/MetaMask/metamask-extension/pull/20363/commits/1ba6cb1f7d489bea929d9b6969e964406b94543e)

File: ui/components/app/incoming-trasaction-toggle/hooks.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (brad-decker) [reply ui/components/app/incoming-trasaction-toggle/hooks.js]: I can only speak from my opinions, but separating logic out into a separate file does not make it easier to read. We used to do this with our component/container pattern. You'd have to constantly switch between files to understand how things connect together. 

Though, a quick check of the hooks/ folder shows many that we are using only in one place. So here's what i'd ask:

Either
A) Inline the hook into the file it is used in. 
B) Move it to the hooks/ folder so that we at least preserve one pattern and we can then discuss how we want to handle this in the future.

File: ui/components/app/incoming-trasaction-toggle/NetworkToggle.js [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+
+import Tooltip from '../../ui/tooltip';
+import {
+  CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP,
+  CHAIN_IDS,
+} from '../../../../shared/constants/network';
+import ToggleButton from '../../ui/toggle-button';
+import { I18nContext } from '../../../contexts/i18n';
+import Identicon from '../../ui/identicon/identicon.container';
+import UrlIcon from '../../ui/url-icon';
```
Comment (georgewrmarshall) [inline ui/components/app/incoming-trasaction-toggle/NetworkToggle.js:25]: If these components are intended to represent networks could we use the `AvatarNetwork` component from the component library. This should be the component we are using moving forward for all network identifiers.
https://metamask.github.io/metamask-storybook/?path=/docs/components-componentlibrary-avatarnetwork--docs

File: ui/pages/settings/experimental-tab/experimental-tab.component.js [hunk 7]
```diff
@@ -239,44 +250,49 @@ export default class ExperimentalTab extends PureComponent {
...
+          marginBottom={3}
         >
           <div className="settings-page__content-item">
             <span>{t('transactionSecurityCheck')}</span>
             <div className="settings-page__content-description">
-              <Typography
-                variant={TypographyVariant.H6}
+              <Text
+                variant={TextVariant.bodySm}
+                as="h6"
```
Comment (georgewrmarshall) [inline ui/pages/settings/experimental-tab/experimental-tab.component.js:272]: In an attempt to lower the bar for entry for our external contributors, I may have unintentionally made it look like the `as` prop was a required prop or "best practice" for the `<Text>` component when it is completely optional. There are many instances of `<Text as="h6"` in the code base which was a result of external contributors swapping the deprecated `<Typography>` with `<Text>` allowing external contributors to make these updates without having to update any e2e tests. I now realize that was a mistake. There probably aren't many times when an `<h6>` element is appropriate. It is totally fine to use the `Text` component without the `as` prop unless it is intentional. Most of the time the default rendered HTML element should be sufficient.

**Review by georgewrmarshall — COMMENTED** [review]
I haven't had a chance to pull your branch and check the UI locally but code is looking great! I have made a couple small suggestions

File: ui/pages/settings/experimental-tab/experimental-tab.component.js (hunk 7 above)
Comment (DDDDDanica) [reply ui/pages/settings/experimental-tab/experimental-tab.component.js:272]: Thanks for the information. I always thought `as` is mandatory ~ will remove now 🙏🏻

File: ui/components/app/incoming-trasaction-toggle/NetworkToggle.js (hunk 6 above)
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/NetworkToggle.js]: Thanks George, this is a great suggestion. I didn't know about this component and it is very smooth to use. Now just adapted to `AvatarNetwork` 👍🏻

**Comment by DDDDDanica** [comment]
Hey @georgewrmarshall Thanks for reviewing, it is a monster MR and i have isolated UI changes in this one 
https://github.com/MetaMask/metamask-extension/pull/20374

This would be easier to review and see changes, 

Thanks a lot in advance 🙏🏻 ❤️

File: app/scripts/controllers/incoming-transactions.test.js [hunk 8]
```diff
@@ -2,7 +2,7 @@ import { strict as assert } from 'assert';
 import sinon from 'sinon';
 import proxyquire from 'proxyquire';
 import nock from 'nock';
-import { cloneDeep } from 'lodash';
+import cloneDeep from 'lodash/cloneDeep';
```
Comment (NidhiKJha) [inline app/scripts/controllers/incoming-transactions.test.js:5]: Why are we changing this. I think we should keep it the same 
```suggestion
import { cloneDeep } from 'lodash';```

File: test/e2e/tests/onboarding.spec.js [hunk 9]
```diff
@@ -162,11 +163,12 @@ describe('MetaMask onboarding', function () {
         );
 
         // Check that the error message is displayed for the password fields
-        const passwordErrorIsDisplayed = await driver.isElementPresent({
-          text: "Passwords don't match",
-          css: 'h6',
-        });
-        assert.equal(passwordErrorIsDisplayed, true);
+        await driver.isElementPresent(
+          // eslint-disable-next-line prettier/prettier
```
Comment (NidhiKJha) [inline test/e2e/tests/onboarding.spec.js:167]: Nit: We can remove this comment

File: ui/pages/settings/experimental-tab/experimental-tab.component.js [hunk 10]
```diff
@@ -6,11 +6,14 @@ import {
...
 import { MetaMetricsEventCategory } from '../../../../shared/constants/metametrics';
-import { Text } from '../../../components/component-library';
+import { Text, Box } from '../../../components/component-library';
 import {
   FontWeight,
   TextColor,
   TextVariant,
+  Display,
+  FlexDirection,
+  JustifyContent,
```
Comment (NidhiKJha) [inline ui/pages/settings/experimental-tab/experimental-tab.component.js:16]: These additions should be in code fencing. It's resulting in the build error
Screenshot 2023-08-03 at 2 16 05 PM

File: test/e2e/tests/settings.spec.js [hunk 11]
```diff
@@ -280,4 +311,77 @@ describe('Settings Search', function () {
...
+          '[data-testid="account-options-menu-button"]',
+        );
+        await driver.clickElement({ text: 'Settings', tag: 'div' });
+        await driver.clickElement({ text: 'Security & privacy', tag: 'div' });
+        await driver.waitForSelector('.settings-page__content__modules');
+
+        // For all networks, incoming transactions configuration should be on
+        await driver.waitForSelector(
+          '[data-testid="incoming-transaction-toggle-enable-all"] > .toggle-button--on',
+        );
```
Comment (NidhiKJha) [inline test/e2e/tests/settings.spec.js:336]: If this toggles on all the toggles, then shouldn't we turn it off before testing the toggle ON feature for other toggles otherwise these tests should be incorrect since the toggle is already ON and we are trying to test if the click toggle them ON, whereas it should be off

https://github.com/MetaMask/metamask-extension/assets/39872794/dc10b836-e27d-468d-94e1-579b2600e412

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Looking really good Danica, Left a couple of comments.

File: ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.component.js [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+
+      <Text variant={TextVariant.bodyMd} color={TextColor.textAlternative}>
+        {t('showIncomingTransactionsInformation')}
+      </Text>
+      <Box
+        marginTop={3}
+        display={Display.Flex}
+        flexDirection={FlexDirection.Row}
+        justifyContent={JustifyContent.spaceBetween}
+        data-testid="incoming-transaction-toggle-enable-all"
```
Comment (NidhiKJha) [inline ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.component.js:53]: If I turn the toggle ON for all the incoming transaction and turn any one toggle OFF then the toggle all button is off as well.
https://github.com/MetaMask/metamask-extension/assets/39872794/aa36335b-f2a2-436f-af11-80c50704db60

File: ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.test.js [hunk 13]
```diff
@@ -0,0 +1,210 @@
+import React from 'react';
+import { fireEvent } from '@testing-library/react';
+import { within } from '@testing-library/dom';
```
Comment (NidhiKJha) [inline ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.test.js:3]: Why are we not using this react testing library ?

 import { within } from '@testing-library/react';
```suggestion
import { within } from '@testing-library/react';
```

File: package.json [hunk 14]
```diff
@@ -405,6 +405,7 @@
     "@storybook/storybook-deployer": "^2.8.16",
     "@storybook/test-runner": "^0.10.0",
     "@storybook/theming": "^7.0.11",
+    "@testing-library/dom": "^9.3.1",
```
Comment (NidhiKJha) [inline package.json:408]: I don't think we need it here because the only case we are using it use within and I think RTL already has it.
```suggestion

```

File: ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.component.js (hunk 12 above)
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.component.js]: Hey Nidhi, thanks for checking it out, and this is expected,
- any of sub toggle is off, toggleAll is off
- only when all of sub toggle is on, toggleAll is on

File: test/e2e/tests/onboarding.spec.js (hunk 9 above)
Comment (DDDDDanica) [reply test/e2e/tests/onboarding.spec.js:167]: oh don't know where did it come from. but removed 🙏🏻

File: app/scripts/controllers/incoming-transactions.test.js (hunk 8 above)
Comment (DDDDDanica) [reply app/scripts/controllers/incoming-transactions.test.js:5]: :cookie this is to reduce the bundle size, you can refer to this article => https://www.blazemeter.com/blog/import-lodash-libraries#:~:text=If%20you%20want%20to%20minimize,function%20by%20its%20own%20module.

File: ui/components/app/incoming-trasaction-toggle/hooks.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/hooks.js]: Solved by inline hooks from another pr https://github.com/MetaMask/metamask-extension/pull/20374

File: test/e2e/tests/settings.spec.js (hunk 11 above)
Comment (DDDDDanica) [reply test/e2e/tests/settings.spec.js]: hey Nidhi, there will be a follow up ticket to remove `toggle all` function, as it was defined after the dev work started. And by then we can forget about toggle all function and won't have much confusion around 🙏🏻

File: ui/pages/onboarding-flow/privacy-settings/privacy-settings.test.js [hunk 15]
```diff
@@ -89,35 +104,6 @@ describe('Privacy Settings Onboarding View', () => {
     expect(setUseAddressBarEnsResolutionStub.mock.calls[0][0]).toStrictEqual(
       false,
     );
-
-    // toggle back to true
```
Comment (DDDDDanica) [inline ui/pages/onboarding-flow/privacy-settings/privacy-settings.test.js:93]: Removed because there's no logic difference to turn and off while using the same setter

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Looks good. But I think we worked on the UI improvement of toggle button in the same PR and it seems it is not aligned to the right for all of these toggles, also I think we need to look at the popup view. We can work on these in a separate PR, wdyt?

Screenshot 2023-08-18 at 3 05 17 PM


https://github.com/MetaMask/metamask-extension/assets/39872794/a6da12ef-f347-4e90-8063-83dcf38e2044

**Comment by DDDDDanica** [comment]
> Looks good. But I think we worked on the UI improvement of toggle button in the same PR and it seems it is not aligned to the right for all of these toggles, also I think we need to look at the popup view. We can work on these in a separate PR, wdyt?
> 
> Screenshot 2023-08-18 at 3 05 17 PM
> 
>  Screen.Recording.2023-08-18.at.3.02.35.PM.mov

Hey @NidhiKJha Good catch. And exactly, we were working on the same page changes within at least 4 of us... it is better to make a separate MR after this one to catch up the rest of UI issues ~

File: test/e2e/tests/settings-toggle.spec.js [hunk 16]
```diff
@@ -0,0 +1,140 @@
...
+          text: 'Confirm',
+          tag: 'button',
+        });
+
+        // should render image now
+        const nftImage = await driver.findElement('[data-testid="nft-image"]');
+        assert.equal(await nftImage.isDisplayed(), true);
+      },
+    );
+  });
```
Comment (NidhiKJha) [inline test/e2e/tests/settings-toggle.spec.js:66]: I think this test should not be in this file?

Comment (DDDDDanica) [reply test/e2e/tests/settings-toggle.spec.js]: Good catch ! removed ~

File: ui/components/ui/toggle-button/index.scss (hunk 2 above)
Comment (georgewrmarshall) [reply ui/components/ui/toggle-button/index.scss:4]: Also not sure we want to center the text for all instances of the `ToggleButton`? Which instance needs to be centered and can it conform to left aligned?

Comment (georgewrmarshall) [reply ui/components/ui/toggle-button/index.scss:4]: Also not sure we want to center the text for all instances of the `ToggleButton`? Which instance needs to be centered and can it conform to left aligned?

File: ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.tsx [hunk 17]
```diff
@@ -88,7 +87,9 @@ const IncomingTransactionToggle = ({
         justifyContent={JustifyContent.spaceBetween}
```
Comment (georgewrmarshall) [inline ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.tsx:87]: We could add a gap prop here so there is some space between the text and toggle button
```suggestion
        justifyContent={JustifyContent.spaceBetween}
        gap={4}
```

**Review by georgewrmarshall — COMMENTED** [review]
LGTM! One small nit is to add a `gap={4}`(16px) gap between the columns of text and toggle buttons so there is some padding when the line length is large or screen is smaller. also `ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.tsx` could use a story so it can be worked on isolation.

- Pulled branch and followed flow from PR description
-

Screenshot 2023-08-18 at 2 33 42 PM
Screenshot 2023-08-18 at 2 34 34 PM

File: app/scripts/metamask-controller.js [hunk 18]
```diff
@@ -416,6 +416,7 @@ export default class MetamaskController extends EventEmitter {
       ),
       tokenListController: this.tokenListController,
       provider: this.provider,
+      networkConfigurations: this.networkController.state.networkConfigurations,
```
Comment (albertolive) [inline app/scripts/metamask-controller.js:419]: @DDDDDanica, I'm unsure, but don't we need the same logic in `mmi-controller.js`?

File: app/scripts/migrations/095.ts [hunk 19]
```diff
@@ -0,0 +1,102 @@
...
+export const version = 95;
+
+/**
+ * This migration will operate the following:
+ *
+ * - Delete `showIncomingTransactions` from `featureFlags` in PreferencesController
+ * - Create a new object under PreferencesController named as `incomingTransactionPreferences`
+ * 1. which will collect all added networks including localhost
+ * 2. then append the test networks
+ * 3. each of them would become a key coming with the value Ture/False from `showIncomingTransactions`
```
Comment (albertolive) [inline app/scripts/migrations/095.ts:19]: `* 3. each of them would become a key coming with the value True/False from `showIncomingTransactions``

Comment (DDDDDanica) [reply app/scripts/migrations/095.ts:19]: I wonder if point out `showIncomingTransactions` this way to indicate it as saved variable is unclear ?

File: ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.tsx (hunk 17 above)
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.tsx:87]: Good catch ! will add to styles !

File: app/scripts/metamask-controller.js (hunk 18 above)
Comment (DDDDDanica) [reply app/scripts/metamask-controller.js:419]: Hey @albertolive i'm not 100% sure about mmi-controller relation with mm-controller.  In this new added feature, we pass in all networks to specify the preferences for each network for selected account. do you think we would need such feature in MMI as well?

File: ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.tsx (hunk 17 above)
Comment (DDDDDanica) [reply ui/components/app/incoming-trasaction-toggle/incoming-transaction-toggle.tsx:87]: @georgewrmarshall also added a storybook 🙏🏻 thanks for the review ❤️

File: app/scripts/metamask-controller.js (hunk 18 above)
Comment (albertolive) [reply app/scripts/metamask-controller.js:418]: Ok, I think it's not needed for now. We'll focus on MetaMerge and then, if we need to do something with this, we'll let you know

**Review by albertolive — DISMISSED** [review]

File: app/scripts/metamask-controller.js (hunk 18 above)
Comment (DDDDDanica) [reply app/scripts/metamask-controller.js:418]: thank you !!

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Looking really good. But when I try to add a new network, then it's added in the list but unfortunately toggle all doesn't work for that. Also, I feel there is a lag with toggle

https://github.com/MetaMask/metamask-extension/assets/39872794/111b9a22-2582-4d92-87db-23778d6b4f93

**Comment by DDDDDanica** [comment]
Hey @NidhiKJha good found !!
Just to make it more clear, for new added network, it would be within this ticket scope https://app.zenhub.com/workspaces/metamask-extension-63529dce65cbb100265a3842/issues/gh/metamask/metamask-planning/1121
and after we started the developing, there is a design change for removing toggle all and we decided to remove to another ticket because it was not included in original scope
https://app.zenhub.com/workspaces/metamask-extension-63529dce65cbb100265a3842/issues/gh/metamask/metamask-planning/1120

I hope after these 2 ticket we would have no issues in toggle all and new added network 🙏🏻

File: app/scripts/migrations/096.ts [hunk 20]
```diff
@@ -0,0 +1,102 @@
...
+  PreferencesController?: {
+    featureFlags?: FeatureFlags;
+    incomingTransactionsPreferences?: IncomingTransactionPreferences;
+  };
+  NetworkController?: {
+    networkConfigurations?: Record<string, NetworkConfiguration>;
+  };
+}
+
+function transformState(state: State) {
```
Comment (Gudahtt) [inline app/scripts/migrations/096.ts:58]: Could you update this to use `Record<string, unknown>` type instead? The use of this type in other migrations was intentional, as it forces us to make zero assumptions about the state shape so that we don't mishandle corrupted state or create corrupted state.

File: app/scripts/migrations/096.ts [hunk 21]
```diff
@@ -0,0 +1,102 @@
...
+    ...addedNetwork,
+    ...testNetworks,
+  ];
+
+  const incomingTransactionsPreferences = fromPairs(
+    map(allSavedNetworks, (element) => [element, showIncomingTransactions]),
+  );
+
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
```
Comment (Gudahtt) [inline app/scripts/migrations/096.ts:95]: For future reference, we have this ESLint error because we should be using `@ts-expect-error` over `@ts-ignore`, because `@ts-expect-error` will tell is when the type error goes away.

Ideally we'd use neither of course, but if we really can't fix a type error, that's how we'd silence it.

In this case, it seems that the type error can and should be fixed, so we don't need either.

**Review by georgewrmarshall — COMMENTED** [review]
Nice work on the story! One suggestion

**Review by albertolive — DISMISSED** [review]

**Review by albertolive — COMMENTED** [review]
Waiting for changes requested :)

File: app/scripts/migrations/096.ts (hunk 21 above)
Comment (DDDDDanica) [reply app/scripts/migrations/096.ts:95]: Thanks for suggestion ! i did some object destructuring and solved the warning 🌟

File: ui/pages/settings/advanced-tab/__snapshots__/advanced-tab.component.test.js.snap [hunk 22]
```diff
@@ -72,6 +72,7 @@ exports[`AdvancedTab Component should match snapshot 1`] = `
     <div
       class="mm-box settings-page__content-row mm-box--display-flex mm-box--flex-direction-row mm-box--justify-content-space-between"
       data-testid="advanced-setting-hex-data"
+      gaps="4"
```
[the remark this answers is not in the document]
Comment (DDDDDanica) [reply ui/pages/settings/advanced-tab/__snapshots__/advanced-tab.component.test.js.snap]: thank you for spotting ! i didn't pay attention either. modified all 🙏🏻

File: app/scripts/migrations/096.ts (hunk 20 above)
Comment (DDDDDanica) [reply app/scripts/migrations/096.ts]: Yes absolutely ! will change here [51f8dda9f2911d242e38a9dc05ad623f570fcd18](https://github.com/MetaMask/metamask-extension/pull/20363/commits/51f8dda9f2911d242e38a9dc05ad623f570fcd18)

**Review by NidhiKJha — APPROVED** [review]
Functionality looks good to me. UI wise we need a follow up PR for the Settings Page

**Review by albertolive — APPROVED** [review]
